### PR TITLE
[it-test] Add examples folder and use in integration test

### PR DESCRIPTION
**Pull request by jaylees14**
Move the `tinder.temple` over (`amazon.temple` didn't generate correctly, will move over when I've worked out why..)

Invoke them from the integration test automatically, so that any new examples are automatically generated and validated.

### CODE REVIEW

**Review by willjvsmith — APPROVED** [review]
Good stuff

**Review by Lewiky — APPROVED** [review]
LGTM

File: src/it/scala/temple/EndpointIntegrationTest.scala [hunk 1]
```diff
@@ -11,4 +13,11 @@ class EndpointIntegrationTest extends EndpointTesterSpec with Matchers {
     val simpleTemple = FileUtils.readFile("src/test/scala/temple/testfiles/simple-dc.temple")
     noException should be thrownBy buildAndTestEndpoints(simpleTemple)
   }
+
+  it should "correctly generate the example Templefiles" in {
+    Files.list(Paths.get("examples")).map(path => path.toString).forEach { (path: String) =>
```
Comment (c-harding) [inline src/it/scala/temple/EndpointIntegrationTest.scala]: Worth filtering for `*.temple` so we can add a readme to that folder without it being imported?

File: examples/tinder.temple [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  #database(postgres);
+  #provider(dockerCompose);
+}
+
+User: service {
+  // Implicit `id` field that matches Auth service `id` field
+  firstName: string;
+  lastName: string;
+  dob: datetime;
+  #auth(email); // Creates Auth service, which stores an `id`, `email` and `password` field
```
Comment (c-harding) [inline examples/tinder.temple]: #319 will break this, race you 😉

**Review by c-harding — APPROVED** [review]

**Pull request merged**